src/commands/amend.rs

```diff
@@ -0,0 +1,128 @@
+//! Amend the current commit.
+//!
+//! This command amends the HEAD commit with changes to files
+//! that are already tracked in the repo. Following the amend,
+//! the command performs a restack.
+
+use std::fmt::Write;
+use std::time::SystemTime;
+
+use eyre::Context;
+use itertools::Itertools;
+
+use crate::commands::gc::mark_commit_reachable;
+use crate::commands::restack;
+use crate::core::config::get_restack_preserve_timestamps;
+use crate::core::effects::Effects;
+use crate::core::eventlog::{Event, EventLogDb};
+use crate::git::{AmendFastOptions, FileStatus, GitRunInfo, Repo};
+use crate::opts::MoveOptions;
+
+/// Amends the existing HEAD commit.
+pub fn amend(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    move_options: &MoveOptions,
+) -> eyre::Result<isize> {
+    let now = SystemTime::now();
+    let repo = Repo::from_current_dir()?;
+    let conn = repo.get_db_conn()?;
+    let mut event_log_db = EventLogDb::new(&conn)?;
+
+    let head_oid = match repo.get_head_info()?.oid {
+        Some(oid) => oid,
+        None => eyre::bail!("No HEAD to amend to"),
+    };
+    let head_commit = repo.find_commit_or_fail(head_oid)?;
+
+    let index = repo.get_index()?;
+    if index.has_conflicts() {
+        writeln!(
+            effects.get_output_stream(),
+            "Cannot amend, because there are unresolved merge conflicts."
+        )?;
+        return Ok(1);
+    }
+
+    let staged_index_paths = repo.get_staged_paths()?;
+    let opts = if !staged_index_paths.is_empty() {
+        let dirty_working_tree = repo.has_changed_files(effects, git_run_info)?;
+        if dirty_working_tree {
+            writeln!(
+                effects.get_output_stream(),
+                "Found staged files in the index. Unstaged changes will not be amended."
```

Comment:
> let's remove this message and add a summary message to immediately after the amend operation took place. It should also contain some brief summary information. I think it would be good to say "Amended with X uncommitted changes"/"Amended with X staged changes (Y uncommitted changes were not amended)."

Great idea; done. Though, I didn't implement the `(Y uncommitted changes were not amended)` portion -- and left it ambiguous as `(Some uncommitted changes were not amended)`, since I think getting the number of modified files in the working copy would require a `git_status` run. I left a TODO to consider adding that in the future.